src/main/java/uk/gov/moj/sdt/service/FeedbackService.java

```diff
@@ -0,0 +1,139 @@
+        } else if (RequestType.WARRANT.getType().equals(requestType)) {
+            updateWarrantRequest(cmcFeedback, individualRequest);
+        } else if (RequestType.JUDGMENT_WARRANT.getType().equals(requestType)) {
+            updateCombinedJudgementWarrantRequest(cmcFeedback, individualRequest);
+        }
+        individualRequestDao.persist(individualRequest);
+    }
+
+    private void updateCreateClaimRequest(CMCFeedback cmcFeedback,
+                                          IIndividualRequest individualRequest) {
+        addStatusAndErrorLog(cmcFeedback, individualRequest);
+        ClaimResponse response = new ClaimResponse();
+        response.setClaimNumber(cmcFeedback.getClaimNumber());
+        response.setIssueDate(cmcFeedback.getIssueDate());
+        response.setServiceDate(cmcFeedback.getServiceDate());
+        response.setFee(cmcFeedback.getFee());
+        individualRequest.setTargetApplicationResponse(convertObjectToXml(response));
+    }
+
+    private void updateJudgementRequest(CMCFeedback cmcFeedback,
+                                          IIndividualRequest individualRequest) {
+        addStatusAndErrorLog(cmcFeedback, individualRequest);
+        JudgementResponse response = new JudgementResponse();
+        response.setJudgmentWarrantStatus(JudgmentWarrantStatus
+                                              .getJudgmentWarrantStatus(cmcFeedback.getJudgmentWarrantStatus()));
+        response.setJudgmentEnteredDate(cmcFeedback.getJudgmentEnteredDate());
+        response.setFirstPaymentDate(cmcFeedback.getFirstPaymentDate());
+        individualRequest.setTargetApplicationResponse(convertObjectToXml(response));
+    }
+
+    private void updateWarrantRequest(CMCFeedback cmcFeedback,
+                                        IIndividualRequest individualRequest) {
+        addStatusAndErrorLog(cmcFeedback, individualRequest);
+        WarrantResponse response = new WarrantResponse();
+        response.setIssueDate(cmcFeedback.getIssueDate());
+        response.setJudgmentWarrantStatus(JudgmentWarrantStatus
+                                              .getJudgmentWarrantStatus(cmcFeedback.getJudgmentWarrantStatus()));
+        response.setFee(cmcFeedback.getFee());
+        response.setWarrantNumber(cmcFeedback.getWarrantNumber());
+        response.setEnforcingCourtCode(cmcFeedback.getEnforcingCourtCode());
+        response.setEnforcingCourtName(cmcFeedback.getEnforcingCourtName());
+        individualRequest.setTargetApplicationResponse(convertObjectToXml(response));
+    }
+
+    private void updateCombinedJudgementWarrantRequest(CMCFeedback cmcFeedback,
+                                      IIndividualRequest individualRequest) {
+        addStatusAndErrorLog(cmcFeedback, individualRequest);
+        JudgementWarrantResponse response = new JudgementWarrantResponse();
+        response.setJudgmentWarrantStatus(JudgmentWarrantStatus.valueOf(cmcFeedback.getJudgmentWarrantStatus()));
```

Comment:
Should this now use the new `getJudgmentWarrantStatus()` method instead of `valueOf()`?

Reply:
done